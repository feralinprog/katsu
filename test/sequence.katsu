use: {
    "core.sequence"
    "core.builtin.misc"
}

let: (seq: Sequence) pprint-all do: [
    print: "length = " ~ seq length >string
    seq each: [ pretty-print: it ]
]

{ 1; 2; 3 } pprint-all
{ 1; 2; 3 } reversed pprint-all
({ 0; 1; 2; 3; 4; 5 } from: 2 to<: 5) pprint-all
({ 0; 1; 2; 3; 4; 5 } from: 1 to<: 4) reversed pprint-all
({ 0; 1; 2; 3; 4; 5 } from: 2 to=: 5) pprint-all

{ "a"; "b"; "c" } each/index: \i x [
    print: i >string ~ " -> " ~ x
]

(3 to<: 6) pprint-all
(3 to=: 6) pprint-all
(6 to=: 3) pprint-all

({ "a"; "b"; "c"; "d"; "e"; "f" } at: { 4; 2; 1; 5 }) pprint-all
