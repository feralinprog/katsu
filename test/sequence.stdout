length = 3
fixnum 1
fixnum 2
fixnum 3
length = 3
fixnum 3
fixnum 2
fixnum 1
length = 3
fixnum 2
fixnum 3
fixnum 4
length = 3
fixnum 3
fixnum 2
fixnum 1
length = 4
fixnum 2
fixnum 3
fixnum 4
fixnum 5
0 -> a
1 -> b
2 -> c
length = 3
fixnum 3
fixnum 4
fixnum 5
length = 4
fixnum 3
fixnum 4
fixnum 5
fixnum 6
length = 0
length = 4
*string: "e"
*string: "c"
*string: "b"
*string: "f"
*vector: length=4 [
  v_array = *array: length=4
    0 = fixnum 1
    1 = fixnum 2
    2 = *string: "c"
    3 = fixnum 4
]
null
0 -> a
1 -> b
2 -> c
abc
a
bool false
abc
a
bool true
bool false
bool true
bool true
bool false
abc
ab
fixnum 2
abc
ab
cd
*string: "nope"
*vector: length=3 [
  v_array = *array: length=4
    0 = *string: "a"
    1 = *string: "b"
    2 = *string: "c"
    3 = null
]
*vector: length=5 [
  v_array = *array: length=5
    0 = *string: "a"
    1 = *string: "b"
    2 = *string: "c"
    3 = *string: "d"
    4 = *string: "e"
]
*vector: length=3 [
  v_array = *array: length=4
    0 = *string: "a"
    1 = *string: "b"
    2 = *string: "c"
    3 = null
]
*vector: length=2 [
  v_array = *array: length=2
    0 = *string: "d"
    1 = *string: "e"
]
*array: length=4
  0 = fixnum 1
  1 = fixnum 2
  2 = fixnum 3
  3 = fixnum 4
*vector: length=6 [
  v_array = *array: length=6
    0 = *string: "a"
    1 = *string: "c"
    2 = *string: "d"
    3 = *string: "e"
    4 = *string: "e"
    5 = *string: "f"
]
*array: length=2
  0 = fixnum 4
  1 = fixnum 5
*vector: length=2 [
  v_array = *array: length=4
    0 = *string: "abcde"
    1 = *string: "abcd"
    2 = null
    3 = null
]
*vector: length=3 [
  v_array = *array: length=4
    0 = fixnum 2
    1 = fixnum 3
    2 = fixnum 4
    3 = fixnum 4
]
*vector: length=0 [
  v_array = *array: length=0
]
*vector: length=1 [
  v_array = *array: length=1
    0 = fixnum 0
]
*vector: length=2 [
  v_array = *array: length=2
    0 = fixnum 0
    1 = fixnum 1
]
*vector: length=3 [
  v_array = *array: length=4
    0 = fixnum 0
    1 = fixnum 1
    2 = fixnum 2
    3 = null
]
*vector: length=4 [
  v_array = *array: length=4
    0 = fixnum 0
    1 = fixnum 1
    2 = fixnum 2
    3 = fixnum 3
]
